Add contributor path page (#77494)
ansible/ansible: docs/docsite/rst/community/contributor_path.rst

****************
Contributor path
****************

> This section describes the contributor journey from the early beginning, through all the stages to becoming a leader who helps shape the future of Ansible. You can use this path as a roadmap for your long-term participation.

Reviewer: ```suggestion
This section describes the contributor journey from the beginning to becoming a leader who helps shape the future of Ansible. You can use this path as a roadmap for your long-term participation.
```

> Any contribution to the project, even a small one, is very welcome and valuable. Any contribution counts, whether it's feedback on an isssue, pull requiest, or  topic, a documentation change, or a coding contribution. When you contribute regularly, your proficiency and judgment in the related area are increasing fast and, along with this, the importance of your presence in the project.

Reviewer: ```suggestion
Any contribution to the project, even a small one, is very welcome and valuable. Any contribution counts, whether it's feedback on an issue, a pull request, a topic or documentation change, or a coding contribution. When you contribute regularly, your proficiency and judgment in the related area increase and, along with this, the importance of your presence in the project.
```

.. contents::
    :local:

Determine your area of interest
=================================

> First, determine areas that are interesting to you, taking into consideration your current experience and experience you'd like to gain. For example, if you use a certain collection, have a look there. See :ref:`how_can_i_help` for more ideas on how to help.

Reviewer: ```suggestion
First, determine areas that are interesting to you. Consider your current experience and what you'd like to gain. For example, if you use a specific collection, have a look there. See :ref:`how_can_i_help` for more ideas on how to help.
```

Reviewer 2: @ugwutotheeshoes thank for the suggestion!
- Let's maybe split the sentence into 2 `First, determine areas that are interesting to you. Consider your current experience and what you'd like to gain.` ?

Reviewer: OK, That's better that way.
Sorry about the late reply too.

Author: I've updated the suggested edit to match this idea. @ugwutotheeshoes- I think in the future you have the ability to edit your own suggested edits as well. Just click `...` to the top right of the suggestion and it should give you the ability to edit. I'll be doing that here now so I can merge the changes. and thanks!

> Find the corresponding the project

Reviewer: ```suggestion
Find the corresponding project
```

Reviewer 2: good catch, thanks!

====================================

These are multiple community projects in the Ansible ecosystem you could contribute to:

- `Ansible Core <https://docs.ansible.com/ansible-core/devel/index.html>`_
- `Collections <https://docs.ansible.com/ansible/latest/user_guide/collections_using.html>`_
- `AWX <https://github.com/ansible/awx>`_
- `Galaxy <https://galaxy.ansible.com/>`_
- `ansible-lint <https://ansible-lint.readthedocs.io/en/latest/>`_
- `Molecule <https://molecule.readthedocs.io/en/latest/>`_

Learn
=====

> The required skillset depends on the area of interest and the project you'll be working on. Remember that the best way to learn is by doing.

Reviewer: "Remember that the best way to learn is by doing."
is this line complete?

Reviewer 2: Looks complete to me, i.e. "Keep it in mind that the best way to learn is by doing."

Specific knowledge for code developers
----------------------------------------

Code development requires the most technical knowledge. Let's sort out what an Ansible developer should learn.

You should understand at least *basics* of the following tools:

- `Python programming language <https://docs.python.org/3/tutorial/>`_
- `Git <https://git-scm.com/docs/gittutorial>`_
- `GitHub collaborative development model through forks and pull requests <https://docs.github.com/en/github/collaborating-with-pull-requests/getting-started/about-collaborative-development-models>`_

You can learn these tools more in depth when working on your first contributions.

Each Ansible project has its own set of contributor guidelines. Familarize yourself with these as you prepare your first contributions.

* :ref:`Ansible Core development <developer_guide>`.
* :ref:`Ansible collectoin development <developing_collections>` and the collection-level contributor guidelines in the collection repository.

Making your first contribution
==============================

> You can find some ideas how you can contribute in :ref:`how_can_i_help` and the `collection repository <https://github.com/ansible-collections/>`_  ``README`` and ``CONTRIBUTING`` files. To make your first experience as smooth as possible, read the repository documentation carefully, then ask the repository maintainers for guidance if you have any questions.

Reviewer: ```suggestion
You can find some ideas on how you can contribute in :ref:`how_can_i_help` and the `collection repository <https://github.com/ansible-collections/>`_  ``README`` and ``CONTRIBUTING`` files. To make your first experience as smooth as possible, read the repository documentation carefully, then ask the repository maintainers for guidance if you have any questions.
```

You can also look for GitHub issues labeled with the ``easyfix``, ``good_first_issue``, and ``docs`` labels.  Add a comment on the GitHub issue to say you are looking at it and to ask for help if you need it.

Continue to contribute
======================

We don't expect everybody to know everything. Start small, think big. When you contribute regularly, your proficiency and judgment in the related area will improve quickly and, along with this, the importance of your presence in the project.

See :ref:`communication` for ways to communicate and engage with the Ansible community, including working group meetings, accessing the Bullhorn news bulletin, and upcoming contributor summits.

Teach others
============

Share your experience with other contributors through `improving documentation <ommunity_documentation_contributions>`, answering question from other contributors and users  on `Matrix/Libera.Chat IRC <communicate>`, giving advice in issues and pull requests, and discussing the `Community Topics <https://github.com/ansible-community/community-topics/issues>`_.

Become a collection maintainer
===============================

> If you are a code contributor to a collection, you can get extended permissions in the repository and become a maintainer. A collection maintainer is a contributor trusted by the community who makes significant and regular contributions to the project and showed themselves as a specialist in the related area. See :ref:`maintainers` for details.

Reviewer: ```suggestion
If you are a code contributor to a collection, you can get extended permissions in the repository and become a maintainer. A collection maintainer is a contributor trusted by the community who makes significant and regular contributions to the project and serves as a specialist in the related area. See :ref:`maintainers` for details.
```

Reviewer 2: i would keep the sentence as is.
`showed themselves` is an important thing here.
We're trying to say that "the person's judgment and activities (like great reviews, for example) show that they are a specialist in the technical area the collection servers. So we can trust their judgment and promote them to a maintainer."

Author: agreed.
